dotnet/samples/Concepts/Agents/ComplexChat_NestedShopper.cs

```diff
@@ -0,0 +1,227 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+using Azure.AI.OpenAI;
+using Microsoft.SemanticKernel;
+using Microsoft.SemanticKernel.Agents;
+using Microsoft.SemanticKernel.Agents.Chat;
+using Microsoft.SemanticKernel.ChatCompletion;
+using Microsoft.SemanticKernel.Connectors.OpenAI;
+using Resources;
+
+namespace Agents;
+
+/// <summary>
+/// Demonstrate usage of <see cref="KernelFunctionTerminationStrategy"/> and <see cref="KernelFunctionSelectionStrategy"/>
+/// to manage <see cref="AgentGroupChat"/> execution.
+/// </summary>
+public class ComplexChat_NestedShopper(ITestOutputHelper output) : BaseTest(output)
+{
+    protected override bool ForceOpenAI => true;
+
+    private const string LeaderName = "Spokesperson";
```

Comment:
Terms are mixed here i.e. Leader vs SpokesPerson

If I understand correctly there will be one user facing agent and a bunch of internal agents

It would be good to make this distinction clear in the naming, agent names could be:

1. PersonalShopperAgent - This is the one the user interacts with and this is a concept users will understand
2. InternalGiftIdeaAgent - This is the agent the PersonalShopper uses to generate gift ideas, using Internal prefix to indicate this Agent doesn't interact with the user
3. InternalGiftReviewerAgent - This is the agent the PersonalShopper uses to review gift ideas, using Internal prefix to indicate this Agent doesn't interact with the user

Reply:
Ok yes, this is very clear.  I like it.  Updated.